drivers/periph_common/cpuid.c

```diff
@@ -31,6 +31,12 @@
 #ifdef CPUID_ADDR
 void cpuid_get(void *id)
 {
+/* gcc 11.2.0 builtin bounds checking raises the following false positive warnings */
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wpragmas" /* silence the CI due to unknown -Wstringop-overread */
+#pragma GCC diagnostic ignored "-Warray-bounds"
+#pragma GCC diagnostic ignored "-Wstringop-overread"
     memcpy(id, (void *)CPUID_ADDR, CPUID_LEN);
```

Comment:
is there still an error if the pointer is casted to e.g., `char[CPUID_LEN]`? that might look less scary than the pragma hammer.

Reply:
```
/home/fabian/forks/RIOT/drivers/periph_common/cpuid.c: In function 'cpuid_get':
/home/fabian/forks/RIOT/drivers/periph_common/cpuid.c:39:12: error: cast specifies array type
   39 |     memcpy((char[CPUID_LEN])id, (void *)CPUID_ADDR, CPUID_LEN);
      | 
```

```
/home/fabian/forks/RIOT/drivers/periph_common/cpuid.c: In function 'cpuid_get':
/home/fabian/forks/RIOT/drivers/periph_common/cpuid.c:39:16: error: cast specifies array type
   39 |     memcpy(id, (char[CPUID_LEN])CPUID_ADDR, CPUID_LEN);
      |                
```

Reply:
This also doesn´t work.
```C
void cpuid_get(uint8_t id[CPUID_LEN]);
```